tools/crm_ticket.c

```diff
@@ -791,72 +808,82 @@ main(int argc, char **argv)
 
             ticket = find_ticket(ticket_id, &data_set);
             if (ticket == NULL) {
-                rc = -ENXIO;
+                CMD_ERR("No such ticket '%s'", ticket_id);
+                exit_code = CRM_EX_NOSUCH;
                 goto bail;
             }
 
             if (ticket->granted) {
                 ticket_warning(ticket_id, "revoke");
-                rc = -EPERM;
+                exit_code = CRM_EX_INSUFFICIENT_PRIV;
                 goto bail;
             }
         }
 
         rc = delete_ticket_state(ticket_id, cib_conn);
+        if (rc != pcmk_ok) {
+            CMD_ERR("Could not clean up ticket: %s", pcmk_strerror(rc));
+        }
+        exit_code = crm_errno2exit(rc);
 
     } else if (modified) {
         if (ticket_id == NULL) {
-            CMD_ERR("Must supply a ticket id with -t\n");
-            rc = -ENXIO;
+            CMD_ERR("Must supply ticket ID with -t");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
         if (attr_value
             && (attr_name == NULL || strlen(attr_name) == 0)) {
-            CMD_ERR("You need to supply an attribute name with the -S command for -v %s\n", attr_value);
-            rc = -EINVAL;
+            CMD_ERR("Must supply attribute name with -S");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
 
         if (attr_name
             && (attr_value == NULL || strlen(attr_value) == 0)) {
-            CMD_ERR("You need to supply a value with the -v option for -S %s\n", attr_name);
-            rc = -EINVAL;
+            CMD_ERR("Must supply attribute value with -v");
+            exit_code = CRM_EX_USAGE;
             goto bail;
         }
```

Comment:
Since multiple couples of  "-S" and "-v" are supported,  I'd probably still keep the above messages specific for the cases that they are not coupled,  maybe like:
"Must supply attribute name with -S for -v %s" and  "Must supply attribute value with -v for -S %s"
?

Reply:
Ah, that makes sense